lib/logstash/filters/util/lookup.rb

```diff
@@ -0,0 +1,156 @@
+require_relative "lookup_result"
+
+module LogStash module Filters module Util
+  class Lookup
+    class Sprintfier
+      def initialize(param)
+        @param = param
+      end
+
+      def fetch(event)
+        event.sprintf(@param)
+      end
+    end
+
+    class Getfier
+      def initialize(param)
+        @param = param
+      end
+
+      def fetch(event)
+        event.get(@param)
+      end
+    end
+
+    attr_reader :target, :query, :parameters
+
+    def self.validate(array_of_options)
+      errors = []
+      array_of_options.each_with_index do |options, i|
+        instance = new(options, {}, Object.new, "lookup-#{i.next}")
+        unless instance.valid?
+          errors << instance.formatted_errors
+        end
+      end
+      return nil if errors.empty?
+      errors.join("; ")
+    end
+
+    def initialize(options, globals, logger, default_id)
+      @target = options["target"]
+      @id = target || default_id
+      @options = options
+      @globals = globals
+      @logger = logger
```

Comment:
I looked at that, the plugin has included the Loggable already and its has the plugin name and the log4j context, so if its used in support classes it will have a new log4j context that would look strange in the logs IMO. So I'll pass the plugin logger object around but its very clunky to do so.